arcgis-ios-sdk-samples/Analysis/Distance measurement analysis/DistanceMeasurementAnalysisViewController.swift

```diff
@@ -0,0 +1,161 @@
+    }
+    
+    @IBAction func unitSystemSegmentedControl(_ sender: UISegmentedControl) {
+        guard let unitSystem = AGSUnitSystem(rawValue: sender.selectedSegmentIndex) else { return }
+        locationDistanceMeasurement.unitSystem = unitSystem
+    }
+    
+    let measurementFormatter: MeasurementFormatter = {
+        let measurementFormatter = MeasurementFormatter()
+        measurementFormatter.numberFormatter.minimumFractionDigits = 2
+        measurementFormatter.numberFormatter.maximumFractionDigits = 2
+        return measurementFormatter
+    }()
+    
+    func updateMeasurementLabels() {
+        guard isViewLoaded else { return }
+        if locationDistanceMeasurement.startLocation != locationDistanceMeasurement.endLocation,
+            let directDistance = locationDistanceMeasurement.directDistance,
+            let horizontalDistance = locationDistanceMeasurement.horizontalDistance,
+            let verticalDistance = locationDistanceMeasurement.verticalDistance {
+            directMeasurementLabel.text = measurementFormatter.string(from: Measurement(distance: directDistance))
+            horizontalMeasurementLabel.text = measurementFormatter.string(from: Measurement(distance: horizontalDistance))
+            verticalMeasurementLabel.text = measurementFormatter.string(from: Measurement(distance: verticalDistance))
+        } else {
+            directMeasurementLabel.text = "--"
+            horizontalMeasurementLabel.text = "--"
+            verticalMeasurementLabel.text = "--"
+        }
+    }
+    
+    // MARK: AGSGeoViewTouchDelegate
+    
+    func geoView(_ geoView: AGSGeoView, didTapAtScreenPoint screenPoint: CGPoint, mapPoint: AGSPoint) {
+        sceneView.screen(toLocation: screenPoint) { [weak self] mapLocation in
+            guard let measurement = self?.locationDistanceMeasurement else { return }
+            if measurement.startLocation != measurement.endLocation {
+                measurement.startLocation = mapLocation
+            }
+            measurement.endLocation = mapLocation
+        }
+    }
+    
+    func geoView(_ geoView: AGSGeoView, didLongPressAtScreenPoint screenPoint: CGPoint, mapPoint: AGSPoint) {
+        sceneView.screen(toLocation: screenPoint) { [weak self] mapLocation in
+            guard let measurement = self?.locationDistanceMeasurement else { return }
+            if measurement.startLocation != measurement.endLocation {
+                measurement.startLocation = mapLocation
+            }
+            measurement.endLocation = mapLocation
+        }
+    }
+    
+    func geoView(_ geoView: AGSGeoView, didMoveLongPressToScreenPoint screenPoint: CGPoint, mapPoint: AGSPoint) {
+        sceneView.screen(toLocation: screenPoint) { [weak self] mapLocation in
+            self?.locationDistanceMeasurement.endLocation = mapLocation
+        }
+    }
+    
+    // MARK: UIViewController
+    
+    override init(nibName nibNameOrNil: String?, bundle nibBundleOrNil: Bundle?) {
```

Comment:
We can dispense with this here for now.

If you think all samples should have this, please open an issue and we can discuss.

Reply:
I copied this from the macOS implementation before the override was removed there. I'll get rid of it.